ludwig/backend/ray.py

```diff
@@ -814,6 +817,36 @@ def initialize(self):
         # Disable placement groups on dask
         dask.config.set(annotations={"ray_remote_args": {"placement_group": None}})
 
+    def _update_dask_backend_with_pg(self, pg):
+        dask.config.set(annotations={"ray_remote_args": {"placement_group": pg}})
+
+    def _clear_dask_backend_pg(self):
+        dask.config.set(annotations={"ray_remote_args": {"placement_group": None}})
+
+    def provision_preprocessing_workers(self):
+        if not self._preprocessor_kwargs.get("use_preprocessing_placement_group", False):
+            logger.warning(
+                "Backend config has use_preprocessing_placement_group set to False or did not set it at all."
+                " provision_preprocessing_workers() is a no-op in this case."
+            )
+            return
+        num_cpu = self._preprocessor_kwargs["num_cpu_workers"]
+        self._preprocessor_pg = placement_group([{"CPU": num_cpu}])
+        ray.get(self._preprocessor_pg.ready())
```

Comment:
Do we need a timeout / failure handling here?

Reply:
added timeout